crates/compiler/src/string_table_manager.rs

```diff
@@ -0,0 +1,79 @@
+//! Adapted from <https://github.com/YarnSpinnerTool/YarnSpinner/blob/da39c7195107d8211f21c263e4084f773b84eaff/YarnSpinner.Compiler/StringTableManager.cs>
+
+use crate::output::StringInfo;
+use std::collections::HashMap;
+use std::ops::{Deref, DerefMut};
+
+#[derive(Debug, Clone, Default)]
+pub(crate) struct StringTableManager(pub HashMap<String, StringInfo>);
```

Comment:
I think I would prefer the HashMap to have a name, as there's at least 5 usages of .0 :)

Reply:
That is the idiomatic way to do it for a newtype. Keep in mind that the .0 is only for explicit calls; a regular called will call `StringTableManager::foo` and it will be deref'ed automatically to `HashMap<_, _>::foo`